rotatetls/rotate-tls.go

```diff
@@ -0,0 +1,233 @@
+		return err
+	}
+
+	fmt.Print("Refreshing controller deployment ...\n")
+	replicas := int32(0)
+	origdeploy.Spec.Replicas = &replicas
+	if _, err := c.K8sClientset.AppsV1().Deployments(namespace).Update(context.Background(), origdeploy, metav1.UpdateOptions{}); err != nil {
+		return err
+	}
+	time.Sleep(10 * time.Second)
+
+	origdeploy, err = c.K8sClientset.AppsV1().Deployments(namespace).Get(context.Background(), deployments.AnnotationsControllerDeploymentName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	origdeploy.Spec.Replicas = deploy.Spec.Replicas
+	if _, err := c.K8sClientset.AppsV1().Deployments(namespace).Update(context.Background(), origdeploy, metav1.UpdateOptions{}); err != nil {
+		return err
+	}
+	time.Sleep(10 * time.Second)
+
+	fmt.Print("Restoring main mutation registation ... \n")
+	if _, err := c.K8sClientset.AdmissionregistrationV1().MutatingWebhookConfigurations().Create(context.Background(), deployments.GetAnnotationsControllerMutationAdmissionConfiguration(namespace, caCert.Bytes()), metav1.CreateOptions{}); err != nil {
+		if !strings.Contains(err.Error(), "already exists") {
+			return err
+		}
+		fmt.Print("KubeArmor Annotation Controller Mutation Admission Registration already exists ...\n")
+	}
+
+	fmt.Print("Deleting temprary ressources ...\n")
+	fmt.Print("Mutation Admission Registration ...\n")
+	if err := c.K8sClientset.AdmissionregistrationV1().MutatingWebhookConfigurations().Delete(context.Background(), mutation.Name, metav1.DeleteOptions{}); err != nil {
+		if !strings.Contains(err.Error(), "not found") {
+			return err
+		}
+		fmt.Print("Mutation Admission Registration not found ...\n")
+	}
+
+	fmt.Print("KubeArmor Annotation Controller Service ...\n")
+	if err := c.K8sClientset.CoreV1().Services(namespace).Delete(context.Background(), service.Name, metav1.DeleteOptions{}); err != nil {
+		if !strings.Contains(err.Error(), "not found") {
+			return err
+		}
+		fmt.Print("KubeArmor Annotation Controller Service not found ...\n")
+	}
+
+	fmt.Print("KubeArmor Annotation Controller Deployment ...\n")
+	if err := c.K8sClientset.AppsV1().Deployments(namespace).Delete(context.Background(), deploy.Name, metav1.DeleteOptions{}); err != nil {
+		if !strings.Contains(err.Error(), "not found") {
+			return err
+		}
+		fmt.Print("KubeArmor Annotation Controller Deployment not found ...\n")
+	}
+
+	fmt.Print("KubeArmor Annotation Controller TLS certificates ...\n")
+	if err := c.K8sClientset.CoreV1().Secrets(namespace).Delete(context.Background(), secret.Name, metav1.DeleteOptions{}); err != nil {
+		if !strings.Contains(err.Error(), "not found") {
+			return err
+		}
+		fmt.Print("KubeArmor Annotation Controller TLS certificates not found ...\n")
+	}
+
+	fmt.Print("Certificates were rotated ...\n")
+	return nil
+}
+
+func getFreeRandPrefix(c *k8s.Client, namespace string) (prefix string, err error) {
+	var found bool
+	for {
+		prefix = randomString(5)
```

Comment:
```suggestion
func getFreeRandSuffix(c *k8s.Client, namespace string) (suffix string, err error) {
	var found bool
	for {
		suffix = randomString(5)
```

~prefix~ suffix since it's being added after the base name 😅

Reply:
hhhh this is the second time this week I make this mistake 😅